.babelrc.js

```diff
@@ -0,0 +1,26 @@
+const { NODE_ENV, BABEL_ENV } = process.env;
+const test = NODE_ENV === 'test';
+const cjs = BABEL_ENV === 'commonjs' || test;
+const loose = true;
+
+module.exports = {
+  presets: [
+    '@babel/typescript',
+    '@babel/react',
+    ['@babel/stage-3', { loose }],
+    ['@babel/env', { modules: false, loose }],
+  ],
+  plugins: [
+    ['@babel/proposal-class-properties', { loose }],
+    '@babel/proposal-pipeline-operator',
+    [
+      'module-resolver',
+      {
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+        root: ['.'],
+      },
+    ],
+    test && './coverage-fix.js',
+    cjs && '@babel/transform-modules-commonjs',
```

Comment:
We should remove this and instead above do:

```
['@babel/env', { modules: cjs, loose }],
```

Or, perhaps more consistently refactor to:

```js
const modules = BABEL_ENV === 'commonjs' || test;
// STUFF
['@babel/env', { modules, loose }],
```


More generally, I try and always steer folks away from taking a dependency in a root `.babelrc` on something that is **not** explicitly in `ROOT/package.json`. And this one isn't -- https://github.com/FormidableLabs/urql/blob/master/package.json#L70-L79

Instead we're rely on `node_modules` flattening which can break in odd cases at odd times because something that works with relying on a flattened location is just "accidentally" working :)

Reply:
I'd used to do this - steering `modules`, but lately I've ditched it, because `transform-modules-commonjs` supports `loose` mode too (notice that I havent used `loose` for separate `transform-modules-commonjs`), but its gain is IMHO minimal, while at the same time it "breaks" spreading namespaced imports - `__esModule` becomes enumerable. I agree that this is not the best pattern in general, but I have found myself doing it from time to time.

Good note on flattening thing though. I can adjust my PR in 2 ways:
- using `modules` option conditionally
- adding `modules-commonjs` as explicit dependency.

Just let me know which one do u prefer

Reply:
Ah, that makes sense. Can you:

- [ ] Add `@babel/transform-modules-commonjs` as a `devDependencies`
- [ ] Capture the gist of your explanation here in a code comment above this line in the `.babelrc` so that we have more context for future folks who pass by this line of code.

?

Thanks!